src/iuse_actor.cpp

```diff
@@ -3091,12 +3090,33 @@ int repair_item_actor::repair_recipe_difficulty( const player &pl,
             cur_difficulty++;
         }
 
-        min = std::min( cur_difficulty, min );
+        min = std::min( cur_difficulty, 5 );
```

Comment:
I believe the original code line was here so if there are several recipes with the matching result, the recipe with the lowest difficulty will be used.

Your code will just use the *last* recipe encountered and all matching recipes encountered before will effectively be ignored. (This unconditionally overrides `min`.)

Reply:
Ah, my bad. I will fix that.